CONTRIBUTING.md

```diff
@@ -111,3 +118,85 @@ and commits.
 ## Reporting Bugs and Creating Issues
 
 When opening a new issue, try to roughly follow the commit message format conventions above.
+
+
+# For developers
+
+## Building DBSP from sources
+
+DBSP is implemented in Rust and uses Rust's `cargo` build system. The SQL
+to DBSP compiler is implemented in Java and uses `maven` as its build system.
+
+You can build the rust sources by runnning the following at the top level of this tree.
+
+```
+cargo build
+```
+
+To build the SQL to DBSP compiler, run the following from `sql-to-dbsp-compiler/SQL-compiler`:
+
+```
+mvn package
+```
+
+If you want to develop DBSP without installing the required toolchains
+locally, you can use Github Codespaces; from
+https://github.com/feldera/dbsp, click on the green `<> Code` button,
+then select Codespaces and click on "Create codespace on main".
+
+## Learning the DBSP rust code
+
+To learn how the DBSP core works, we recommend starting with the tutorial.
+
+From the project root:
+
+```
+cargo doc --open
+```
+
+Then search for `dbsp::tutorial`.
```

Comment:
We should really publish the docs and link to them.
Maybe not now.
Can't you put here the exact location?

Reply:
I couldn't find a way to pass an exact argument to cargo doc :(